### PR TITLE
Jupyter Notebook Tutorial for zero shot text classification

### LINKED ISSUE #734: jupyter notebook classification example with banking77 data
**Describe the bug**

- this issue is part of https://github.com/asyml/forte/issues/691

- this PR partially corrects https://github.com/asyml/forte/issues/691

current jupyter notebook can be found https://github.com/bhaskar2443053/forte/blob/master/docs/notebook_tutorial/Text_classification_Banking77.ipynb

**To Reproduce**
1. Prepare data `...`
2. Run command '....'
3. See error

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - Python and Package verions: [e.g. Python version, Pytorch version]

**Additional context**

### PR DESCRIPTION
**Pull request by bhaskar2443053**
This issue is part of #691 
This PR fixes #734 

### Description of changes
This tutorial is an introduction to the Zero-Shot Learning technique for text classification. 

### Possible influences of this PR.
removed [text_classification_pipeline.ipynb](https://github.com/asyml/forte/blob/master/docs/notebook_tutorial/text_classification_pipeline.ipynb) , as its updated with a new tutorial
### Test Conducted
Describe what test cases are included for the PR.

### CODE REVIEW

**Review by hunterhector — COMMENTED** [review]
Since the tutorial is quite large let me comment them here instead of inline:
1. The rendering format is a bit off, you can check it here: https://asyml-forte--799.org.readthedocs.build/en/799/notebook_tutorial/zero_shot_text_classification.html, notice that the bullet list format is not working
2. The explanation of the pipeline is a bit too rigid. You introduce the whole configuration list, but that's a bit too lengthy and we should probably talk about the code, not what the configuration parameters are.
3. There are grammar errors from time to time, like `from hugging face model`, it should be `from the HuggingFace model`. Instead of `Zero shot is a powerful tool`, it should be `Zero Shot model is a powerful tool`
4. The cell output is really long, I don't think we want to include all those.

Overall, there is an improvement to the writing, but it still does not feel like a tutorial to me.

File: docs/notebook_tutorial/zero_shot_text_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,2652 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c371c5c4",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "Zero shot text classification is an extreme example of transfer learning, where the model tries to predict without any fine tuning. This kind of model predict classification result based on similarity between the input text and label name, so they don't require any labeled data. Well defined Label names are sufficient to achieve a reasonably performing model. We will be using [valhalla/distilbart-mnli-12-1](https://huggingface.co/valhalla/distilbart-mnli-12-1) as our text classification model, which was trained on [Multi-nli](https://huggingface.co/datasets/multi_nli) dataset to make predictions on a subset of amazon review sentiment [ARS](https://s3.amazonaws.com/fast-ai-nlp/amazon_review_polarity_csv.tgz) dataset and [Banking77](https://raw.githubusercontent.com/PolyAI-LDN/task-specific-datasets/master/banking_data/test.csv) dataset. we will also see how the forte pipeline works seamlessly with different third party tools like [nltk](https://www.nltk.org/) and [huggingface](https://huggingface.co/) to make our life easier."
```
Comment (feipenghe) [inline docs/notebook_tutorial/zero_shot_text_classification.ipynb:19]: Could you check grammar errors? I have seen some apparent grammar errors.

File: data_samples/amazon_review_polarity_csv/amazon_sample_4.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+label,Title,description
```
[the remark this answers is not in the document]
Comment (feipenghe) [reply data_samples/amazon_review_polarity_csv/amazon_sample_4.csv:1]: Or just keep a few samples (~20) is good enough for the tutorial.

File: docs/spelling_wordlist.txt [hunk 3]
```diff
@@ -156,3 +156,5 @@ GroupType
 PathLike
 ElementType
 customizable
+sklearn
```
Comment (feipenghe) [inline docs/spelling_wordlist.txt:159]: for package name or suffix, maybe you can mention them by ``sklearn`` and ``csv`` rather than add them into spelling wordlist.

File: data_samples/amazon_review_polarity_csv/amazon_sample_4.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (bhaskar2443053) [reply data_samples/amazon_review_polarity_csv/amazon_sample_4.csv]: removed the 10k sample and added a 20 rows sample.

File: docs/notebook_tutorial/zero_shot_text_classification.ipynb (hunk 1 above)
Comment (bhaskar2443053) [reply docs/notebook_tutorial/zero_shot_text_classification.ipynb]: I fixed using Grammarly.

File: docs/spelling_wordlist.txt (hunk 3 above)
Comment (bhaskar2443053) [reply docs/spelling_wordlist.txt]: used double backquotes to fix spellings not in wordlist

**Comment by bhaskar2443053** [comment]
> 

I am unable to fix the bullet points. I tried '- ' , '* ' and '1. ' , but none of them are rendering correctly in docs build.
any suggestions?

**Comment by hepengfe** [comment]
> I am unable to fix the bullet points. I tried '- ' , '* ' and '1. ' , but none of them are rendering correctly in docs build.
> any suggestions?

According to https://buildmedia.readthedocs.org/media/pdf/nbsphinx/latest/nbsphinx.pdf,
> In case that your Bulletpoints do render in the notebook and do not render with nbsphinx, please add one blank line before the bulletpoints.

You can google the notebook issue with keyword `nbsphinx` which is the package we are using to convert the notebook to html.

**Comment by hepengfe** [comment]
I have a few suggestions.

1. You could remove square brackets around the mentioned issues in the PR description.
2. Some descriptions such as the one below is inaccurate or don't relate very well to forte. I would suggest refer to some official definition and shows how forte could handle data and organize nlp components for this particular task.

> Text classification is the process of grouping text data into various categories based on some algorithm. These algorithms are at the heart of a variety of software systems that process text data at scale


3. The description below is wrong.
> If you want to use CPU, set the value to “-1”.

4. maybe backtick the pipeline component in the markdown cell such as `NLTKSentenceSegmenter( )`

In general, I would suggestion review both notebook and the rendered html page after you are done with editing.

**Comment by bhaskar2443053** [comment]
> > I am unable to fix the bullet points. I tried '- ' , '* ' and '1. ' , but none of them are rendering correctly in docs build.
> > any suggestions?
> 
> According to https://buildmedia.readthedocs.org/media/pdf/nbsphinx/latest/nbsphinx.pdf,
> 
> > In case that your Bulletpoints do render in the notebook and do not render with nbsphinx, please add one blank line before the bulletpoints.
> 
> You can google the notebook issue with keyword `nbsphinx` which is the package we are using to convert the notebook to html.

adding a blank line above bullets worked.

**Comment by bhaskar2443053** [comment]
> 3\. backtick

Mostly fixed all the above comments.

**Comment by hepengfe** [comment]
Some minor issues. After these are fixed I think it's ready to merge.

> These algorithms are at the heart of various software systems that process text data at scale. 
I think we probably don't want to keep this sentence as it doesn't relate very well to either the context or forte.


> body is the complete description of the review.
`body` -> `Body`

> where the model tries to classify without any fine-tuning

without any fine-tuning on a single data.


> let’s also check the class balance of the data by using groupby and count functions in pandas.
`let` -> `Let`

> The first entry to our pipeline will be the data reading step.
Rather than `entry`, we want to refer them as `component` or [`pipeline component`](https://github.com/asyml/forte/blob/53567aebbafbcc40c1b817d9e6621bc894db5a87/forte/pipeline_component.py#L28). As entry is referring to [something else](https://github.com/asyml/forte/blob/c2e6ca5d0ce472880c9d742f4ef92d654f5e75c7/forte/data/ontology/core.py#L156) in forte.


> Accuracy of 95% on unseen data is an exceptional performance from the Hugging-Face model. Zero-Shot learning is a powerful tool for low volume of label data.

I think we don't need to backtick HuggingFace and zero-shot learning as they are already known names. By backtick package names, I mean suppose `transformers` as a python package name shown in markdown cell, we definitely need to backtick it.

**Comment by bhaskar2443053** [comment]
> Some minor issues. After these are fixed I think it's ready to merge.
> 
> > These algorithms are at the heart of various software systems that process text data at scale.
> 
> I think we probably don't want to keep this sentence as it doesn't relate very well to either the context or forte.
> 
> > body is the complete description of the review.
> 
> `body` -> `Body`
> 
> > where the model tries to classify without any fine-tuning
> 
> without any fine-tuning on a single data.
> 
> > let’s also check the class balance of the data by using groupby and count functions in pandas.
> 
> `let` -> `Let`
> 
> > The first entry to our pipeline will be the data reading step.
> 
> Rather than `entry`, we want to refer them as `component` or [`pipeline component`](https://github.com/asyml/forte/blob/53567aebbafbcc40c1b817d9e6621bc894db5a87/forte/pipeline_component.py#L28). As entry is referring to [something else](https://github.com/asyml/forte/blob/c2e6ca5d0ce472880c9d742f4ef92d654f5e75c7/forte/data/ontology/core.py#L156) in forte.
> 
> > Accuracy of 95% on unseen data is an exceptional performance from the Hugging-Face model. Zero-Shot learning is a powerful tool for low volume of label data.
> 
> I think we don't need to backtick HuggingFace and zero-shot learning as they are already known names. By backtick package names, I mean suppose `transformers` as a python package name shown in markdown cell, we definitely need to backtick it.

modified according to the above feedbacks.

**Review by hepengfe — APPROVED** [review]
